### PR TITLE
Swift 3 Update

### LINKED ISSUE #619: Update to Xcode 8 / Swift 3

### PR DESCRIPTION
**Pull request by ashfurrow**
- [x] Migrate syntax
- [x] Fix compiler errors
- [x] Fix unit test compiler errors
- [x] Update to RxSwift RC
- [x] ~~Fix "Use Artsy Login Details" -> "Use Artsy Bidder Details" problem.~~(Opened https://github.com/artsy/eidolon/issues/621 )
- [x] Fix bid preview view being out of place during fulfillment. 
- [x] Get unit tests to pass
- [x] Test on device
- [ ] Test card reader
- [x] Bid price cut off by currency text on first screen of fulfillment
- [x] Fix Fulfillment UI
- [x] Action doesn't appear to be working for conditions of sale
- [x] Test compiler optimizations
- [x] Address deprecation warnings
- [x] ~~Storyboard ID update~~
- [x] "No Internet Connection" screen persists after internet connection returns
- [x] Cursor on bid entry screen is misaligned
- [x] ~~Outbid flow cannot be cancelled _or_ something is wrong with bidding via artsy credentials~~

Fixes #619.

### CODE REVIEW

**Comment by ashfurrow** [comment]
> Executed 246 tests, with 70 failures

obama-not-bad

**Comment by ashfurrow** [comment]
Okay, down to 22 failing tests, 19 of which are snapshot tests.

**Comment by orta** [comment]
Cool 👍

File: Kiosk/App/AppDelegate+GlobalActions.swift [hunk 1]
```diff
@@ -95,13 +95,13 @@ extension AppDelegate {
...
         return CocoaAction { _ in
-            self.hideAllTheThings().then(self.showWebController("https://artsy.net/privacy"))
+            self.hideAllTheThings().then(self.showWebController(address: "https://artsy.net/privacy"))
         }
     }
 
     func showConditionsOfSaleCommand() -> CocoaAction {
         return CocoaAction { _ in
-            self.hideAllTheThings().then(self.showWebController("https://artsy.net/conditions-of-sale"))
+            self.hideAllTheThings().then(self.showWebController(address: "https://artsy.net/conditions-of-sale"))
```
Comment (ashfurrow) [inline Kiosk/App/AppDelegate+GlobalActions.swift]: So the `hideAllTheThings()` observables is dismissing the fulfillment nav controller, which ultimately causes the button who owns this action to be deallocated, and the Action gets deallocated, and the subscription to this observable chain gets disposed of before the web controller has been presented. Yikes.

**Comment by ashfurrow** [comment]
Doing one last on-device check before we merge 🎉

**Pull request merged**